ex_smolvlm.py

```diff
@@ -1,37 +1,89 @@
 import requests
-import json
 import os
+import base64
+import logging
+import argparse
 
+logging.basicConfig(level=logging.INFO)
 
-ip_vllm = os.getenv('MY_URL','http://localhost:8000') #url declared as an env var, if empty, takes localhost:8000 by default
 
-model = os.getenv('MY_MODEL','HuggingfaceTB/SmolVLM-256M-Instruct') #model declared as an env var
+path = '/app/images'
+logging.info("Path")
+
+list_images = [file for file in os.listdir(path) if file.endswith((".jpg"))]
+logging.info("List of images")
+
+
+def image_to_base64(image):
+    """
+    Encodes an image to base64.
+    Params:
+        image: name of the image
+    Returns:
+        string of the image encoded in base64
+    """
+    try:
+        with open(image, "rb") as image:
+            image_bin = image.read()
+        image_base64 = base64.b64encode(image_bin).decode("utf-8")
+        return image_base64
+    except Exception as e:
+        print(e)
+
+list_images_base64 = list(map(image_to_base64,list_images))
+logging.info("List of encoded images")
```

Comment:
Crucial: You are logging some messages but without the content of relevant variables. In this case, I would output also the content of list_images_base64.